docs/kuneiform/introduction.mdx

```diff
@@ -31,18 +27,61 @@ table users {
     address text notnull unique // wallet address
 }
 
-// a public action for creating a user
+// a public procedure for creating a user
 // it will store the user's wallet address
-action create_user ($id, $username, $age) public {
+procedure create_user ($id int, $username text, $age int) public {
     INSERT INTO users
-    VALUES ($id, $username, $age, @caller);
+    VALUES ($id, $username, $age, @caller); // @caller is the wallet address of the calling user
 }
 
-// a public action for updating a user's own mutable data
-action update_user ($username, $age) public {
-    UPDATE users
-    SET username=$username, age=$age
-    WHERE address=@caller;
+// a public, read-only procedure for getting all users
+procedure get_users() public view returns table(username text, age int) {
+    return SELECT username, age FROM users;
+}
+```
+
+## Data Types
+
+Below is a list of all supported data types in Kuneiform.
+
+| Data Type | Description |
+| --------- | ----------- |
+| `int` | A 64-bit integer. |
+| `text` | A string of text. |
+| `bool` | A boolean value. |
+| `blob` | A binary large object. |
+| `uuid` | A 128-bit universally unique identifier. |
+| `decimal(precision, scale)` | A fixed-point number, that can be defined with a precision and scale between 1 and 1000. |
+| `uint256` | A 256-bit unsigned integer. |
+
+ Kuneiform also supports 1-dimensional arrays for all data types. To declare an array, simple include square brackets:
+
+```typescript
+   table users {
+       id int primary,
+       names text[] notnull
+   }
+```
+
+## Contextual Variables
+
+Kuneiform supports "contextual variables", which allow you to access information about the current transaction.
+Contextual variables can be accessed anywhere regular variables can be accessed, and are prefixed with an `@` symbol.
```

Comment:
Should we specify **where** regular variables can be accessed?

Reply:
I don't really know how... if they are curious about how to access variables, they should just look at the procedure language docs.

Reply:
Like I am not sure how I would convey where variables can be used without just showing them how to use the language